Bottom: bbcbbfd7f381c56cf598569a87fad7b423a8356b
Top:    5583b0db9080630fcd95db0343fcc8b6120d574d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-24 19:32:55 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index d5d08f1..feed3b6 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -306,6 +306,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
